pkg/v22/errors/error.go

```diff
@@ -0,0 +1,73 @@
+package errors
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	chartConfigNotInstalledErrorText = "the server could not find the requested resource (get chartconfigs.core.giantswarm.io)"
+)
+
+var (
+	chartConfigNotAvailablePatterns = []*regexp.Regexp{
+		regexp.MustCompile(`[Get|Patch|Post] https://api\..*/apis/core.giantswarm.io/v1alpha1/namespaces.* (unexpected )?EOF`),
+		regexp.MustCompile(`[Get|Patch|Post] https://api\..*/apis/core.giantswarm.io/v1alpha1/namespaces.* net/http: (TLS handshake timeout|request canceled).*?`),
+	}
+)
+
+// ChartConfigNotAvailableError is returned when the chartconfig custom
+// resources are not available in the tenant cluster.
+var ChartConfigNotAvailableError = &microerror.Error{
+	Kind: "ChartConfigNotAvailableError",
+}
+
+// ChartConfigNotInstalledError is returned when the chartconfig CRD is not
+// installed in a tenant cluster.
+var ChartConfigNotInstalledError = &microerror.Error{
+	Kind: "ChartConfigNotInstalledError",
+}
+
+// IsChartConfigNotAvailable asserts ChartConfigNotAvailableError.
+func IsChartConfigNotAvailable(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	c := microerror.Cause(err)
+
+	for _, re := range chartConfigNotAvailablePatterns {
+		matched := re.MatchString(c.Error())
+
+		if matched {
+			return true
+		}
+	}
+
+	if c == ChartConfigNotAvailableError {
+		return true
+	}
+
+	return false
+}
+
+// IsChartConfigNotInstalled asserts ChartConfigNotInstalledError
+func IsChartConfigNotInstalled(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	c := microerror.Cause(err)
+
+	if strings.Contains(c.Error(), chartConfigNotInstalledErrorText) {
+		return true
+	}
+
+	if c == ChartConfigNotInstalledError {
```

Comment:
I think this statement should be processed earlier than line 64.

Reply:
I just based this on https://github.com/giantswarm/errors/blob/2640113be13f2eec85fe785ed74418a1a2852e5d/tenant/error.go and I think the approach there is correct.

AIUI this is a fallback so if you use `microerror.Maskf(ChartConfigNotInstalledError, "..."` it still matches.

Reply:
@tomahawk28 Does that make sense? PTAL

Reply:
Yes of course, but I thought that the comparing instance type would be faster than the actual string comparison. I think we could change the sequence in this file actually. 	https://github.com/giantswarm/errors/blob/2640113be13f2eec85fe785ed74418a1a2852e5d/tenant/error.go#L53-L55 WDYT? (I would do it after this PR if it makes sense)

Reply:
> Yes of course, but I thought that the comparing instance type would be faster than the actual string comparison.

It may be faster but it will likely only be milliseconds. I don't think that level of optimization is needed. But if you feel this is important and want to do it then go ahead.

But yes lets please do that as a follow up PR as this is blocking the app CR migration.